### ISSUE TITLE
Implementing React Router

### ISSUE THREAD

**Issue by V-FOR-VEND3TTA** [V-FOR-VEND3TTA/bits-and-bytes#5]
React Router is a widely used library for handling routing in React applications. It allows you to create single-page applications (SPAs) by managing the navigation and rendering of different components based on the URL. Here are some reasons why you might need React Router:

1. **Client-Side Routing**: React Router enables client-side routing, meaning that navigation within your application occurs without making a request to the server for each page change. This results in faster and smoother user experiences as only the necessary components are updated, without a full page reload.

   ```jsx
   import { BrowserRouter as Router, Route } from 'react-router-dom';

   function App() {
     return (
       <Router>
         <Route path="/" exact component={Home} />
         <Route path="/about" component={About} />
         <Route path="/contact" component={Contact} />
       </Router>
     );
   }
   ```

2. **Nested Routing**: React Router allows you to nest routes within one another, providing a hierarchical structure for your application. This is useful for creating complex layouts or dividing sections of your app into smaller components.

   ```jsx
   function App() {
     return (
       <Router>
         <Route path="/" exact component={Home} />
         <Route path="/dashboard" component={Dashboard} />
       </Router>
     );
   }
   ```

3. **Dynamic Routing**: React Router supports dynamic routing, which allows you to handle URL parameters and pass them as props to your components. This is valuable for building pages that rely on specific data, such as user profiles.

   ```jsx
   <Route path="/user/:id" component={UserProfile} />
   ```

4. **Linking and Navigation**: React Router provides a `<Link>` component for creating navigation links that update the URL without triggering a full page refresh. This keeps your application's navigation smooth and interactive.

   ```jsx
   import { Link } from 'react-router-dom';

   function Navigation() {
     return (
       <nav>
         <Link to="/">Home</Link>
         <Link to="/about">About</Link>
         <Link to="/contact">Contact</Link>
       </nav>
     );
   }
   ```

5. **404 Error Handling**: You can easily implement a custom 404 error page for handling routes that don't match any defined route. This ensures a better user experience by providing a clear message when users enter an invalid URL.

   ```jsx
   <Route component={NotFound} />
   ```

6. **Code Splitting**: React Router can be used in conjunction with tools like React's lazy loading and Suspense to enable code splitting. This means that different route components are loaded only when they are needed, reducing the initial load time of your application.

   ```jsx
   import { lazy, Suspense } from 'react';

   const About = lazy(() => import('./About'));

   function App() {
     return (
       <Router>
         <Suspense fallback={<div>Loading...</div>}>
           <Route path="/about" component={About} />
         </Suspense>
       </Router>
     );
   }
   ```

In summary, React Router is a powerful library for managing routing in React applications, and it plays a crucial role in creating modern single-page applications. Whether you're building a simple website or a complex web app, React Router can help you handle navigation, organize your code, and enhance the user experience.
